client/state/preferences/actions.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_REMOVE,
+	PREFERENCES_FETCH
+} from 'state/action-types';
+
+export const USER_SETTING_KEY = 'calypso_preferences';
+
+export function fetchPreferences() {
+	return ( dispatch ) => {
+		dispatch( { type: PREFERENCES_FETCH } );
+		return wpcom.me().settings().get().then( ( data ) => {
+			dispatch( {
+				type: PREFERENCES_RECEIVE,
+				data
+			} );
+		} );
+	};
+}
+
+const save = () => ( dispatch, getState ) => {
```

Comment:
I think we should forgo the `save` action, and simply issue a network request when `setPreference` or `removePreference` is called. What would be the use-case here? Even if we wanted to update multiple settings simultaneously, I think a `setPreferences( preferences )` action creator would be preferable.

_(I also think as much as possible, we should avoid `getState` in action creators. It feels a bit too far magic and harms readability as far as understanding the effect it has on the dispatched action, vs. more functional input/output purity)_